test/hotspot/jtreg/compiler/c2/irTests/TestPadding.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+package compiler.c2.irTests;
+
+import compiler.lib.ir_framework.*;
+
+/*
+ * @test
+ * @bug 8309978
+ * @summary [x64] Fix useless padding
+ * @library /test/lib /
+ * @requires vm.compiler2.enabled
+ * @requires (os.simpleArch == "x64")
+ * @run driver compiler.c2.irTests.TestPadding
+ */
+
+public class TestPadding {
+    public static void main(String[] args) {
+        TestFramework.runWithFlags("-XX:+IntelJccErratumMitigation");
+    }
+
+    @Run(test = "test")
+    public static void test_runner() {
+        tpf = new TestPadding();
+        for (int i = 0; i < 11000; i++) {
+            test(i);
+            tpf.b1++; // to take both branches in test()
+        }
```

Comment:
`test_runner()` will be invoked 2000 times (default warm-up) before compilation of `test()`. So, this loop will run 2000 * 11000 times. Do you need that many iterations or can the loop be removed such that we only have 2000 warm-up iterations? I.e. something like:
```
    @Run(test = "test")
    public static void test_runner() {
        tpf = new TestPadding();
        test(42);
        tpf.b1++; // to take both branches in test()

    }
```
If you need more iterations, you could still specify `@Warmup(12345)` at `test_runner()` to get more profiling in before compilation of `test()`.

Reply:
That looks good to me, thanks for the update!